crates/proof-of-sql/src/base/database/column_operation.rs

```diff
@@ -0,0 +1,202 @@
+    numbers_of_smaller_type: Vec<SmallerType>,
+    numbers_of_larger_type: Vec<LargerType>,
+) -> Vec<LargerType>
+where
+    SmallerType: Copy,
+    LargerType: From<SmallerType> + Add<Output = LargerType> + Copy,
+{
+    numbers_of_smaller_type
+        .into_iter()
+        .zip(numbers_of_larger_type)
+        .map(|(l, r)| LargerType::from(l) + r)
+        .collect()
+}
+
+/// Add two numerical vectors as decimals.
+fn add_decimal_columns<S, T0, T1>(
+    lhs: Vec<T0>,
+    rhs: Vec<T1>,
+    left_precision_value: u8,
+    left_scale: i8,
+    right_precision_value: u8,
+    right_scale: i8,
+) -> ColumnOperationResult<OwnedColumn<S>>
+where
+    S: Scalar + core::convert::From<T0> + core::convert::From<T1>,
+    T0: Copy,
+    T1: Copy,
+{
+    let new_scale = left_scale.max(right_scale);
+    let left_upscale = new_scale - left_scale;
+    let right_upscale = new_scale - right_scale;
+    let new_precision_value: i16 = new_scale as i16
+        + (left_precision_value as i16 - left_scale as i16)
+            .max(right_precision_value as i16 - right_scale as i16)
+        + 1_i16;
+    let new_precision = u8::try_from(new_precision_value)
+        .map_err(|_| {
+            ColumnOperationError::DecimalConversionError(DecimalError::InvalidPrecision(
+                new_precision_value.to_string(),
+            ))
+        })
+        .and_then(|p| {
+            Precision::new(p).map_err(|_| {
+                ColumnOperationError::DecimalConversionError(DecimalError::InvalidPrecision(
+                    p.to_string(),
+                ))
+            })
+        })?;
+    let scalars: Vec<S> = lhs
+        .into_iter()
+        .zip(rhs)
+        .map(|(l, r)| -> ColumnOperationResult<S> {
+            Ok(scale_scalar(S::from(l), left_upscale)? + scale_scalar(S::from(r), right_upscale)?)
+        })
+        .collect::<ColumnOperationResult<Vec<S>>>()?;
+    Ok(OwnedColumn::Decimal75(new_precision, new_scale, scalars))
+}
+
+impl<S: Scalar> Add for OwnedColumn<S> {
+    type Output = ColumnOperationResult<Self>;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        if self.len() != rhs.len() {
+            return Err(ColumnOperationError::DifferentColumnLength(
+                self.len(),
+                rhs.len(),
+            ));
+        }
+        match (self.clone(), rhs.clone()) {
+            (Self::SmallInt(lhs), Self::SmallInt(rhs)) => Ok(Self::SmallInt(add_vecs(lhs, rhs))),
+            (Self::SmallInt(lhs), Self::Int(rhs)) => Ok(Self::Int(add_vecs_with_casting(lhs, rhs))),
+            (Self::SmallInt(lhs), Self::BigInt(rhs)) => {
+                Ok(Self::BigInt(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::SmallInt(lhs), Self::Int128(rhs)) => {
+                Ok(Self::Int128(add_vecs_with_casting(lhs, rhs)))
+            }
+            (Self::SmallInt(lhs), Self::Decimal75(precision, scale, rhs)) => {
+                add_decimal_columns(lhs, rhs, 5, 0, precision.value(), scale)
```

Comment:
Either this is wrong, or the one below is.
```suggestion
                add_decimal_columns(lhs, rhs, 0, 5, precision.value(), scale)
```

Reply:
Nah. Precision before scale.

Reply:
Right. See my other comment.